src/android/CHIPTool/app/src/main/java/com/google/chip/chiptool/clusterclient/OnOffClientFragment.kt

```diff
@@ -110,39 +107,11 @@ class OnOffClientFragment : Fragment() {
   }
 
   private fun updateAddressClick() {
-    val serviceInfo = NsdServiceInfo().apply {
-      serviceName = "%016X-%016X".format(
-        fabricIdEd.text.toString().toLong(),
-        deviceIdEd.text.toString().toLong()
-      )
-      serviceType = "_matter._tcp"
-    }
-
-    // TODO: implement the common CHIP mDNS interface for Android and make CHIP stack call the resolver
-    val resolverListener = object : NsdManager.ResolveListener {
-      override fun onResolveFailed(serviceInfo: NsdServiceInfo?, errorCode: Int) {
-        showMessage("Address resolution failed: $errorCode")
-      }
-
-      override fun onServiceResolved(serviceInfo: NsdServiceInfo?) {
-        val hostAddress = serviceInfo?.host?.hostAddress ?: ""
-        val port = serviceInfo?.port ?: 0
-
-        showMessage("Address: ${hostAddress}:${port}")
-
-        if (hostAddress == "" || port == 0)
-          return
-
-        try {
-          deviceController.updateAddress(deviceIdEd.text.toString().toLong(), hostAddress, port)
-        } catch (e: ChipDeviceControllerException) {
-          showMessage(e.toString())
-        }
-      }
-    }
-
-    (requireContext().getSystemService(Context.NSD_SERVICE) as NsdManager).apply {
-      resolveService(serviceInfo, resolverListener)
+    try{
+      deviceController.updateDevice(fabricIdEd.text.toString().toLong(), deviceIdEd.text.toString().toLong())
```

Comment:
How will this behave if the value is outside the signed-64-bit range (i.e. > 2^63)?

Reply:
Right, I guess I should have used `toULong` here. Will fix.

Reply:
Fixed.